server/src/main/java/org/elasticsearch/plugins/PluginsService.java

```diff
@@ -367,6 +368,9 @@ private static Bundle readPluginBundle(final Set<Bundle> bundles, final Path plu
         if (bundles.add(bundle) == false) {
             throw new IllegalStateException("duplicate " + type + ": " + info);
         }
+        if (type.equals("module") && info.getName().startsWith("test-") && Build.CURRENT.isSnapshot() == false) {
```

Comment:
Seems basing this on name is a little brittle. Is there no way to add arbitrary metadata to modules?

Reply:
There is not. We can add the the plugin properties, but I did not want to add an "official" property here, as it is really only about modules, yet the properties apply to all plugins and modules. I think the name checking is fine for now, since we completely own the namespace of modules.